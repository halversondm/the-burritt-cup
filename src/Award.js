import React from 'react';
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';

class Award extends React.Component {

    render() {
        return (
            <div>
                <hr/>
                <Row className="justify-content-center">
                    <Col lg={8} md={10}>
                        <div className="post-preview">
                            {this.props.image !== "" ?
                                <img className="img-responsive" src={this.props.image} alt={this.props.alt}/> : null}
                            <h2 className="post-title">{this.props.title}</h2>
                            <h4 className="post-subtitle">{this.props.subtitle}</h4>
                            <p className="post-meta">{this.props.metatitle}</p>
                            <p className="post-meta">Past Winners:</p>
                            <table className="table table-striped table-responsive">
                                <thead>
                                    <tr>
                                        <th>Year</th>
                                        <th>Players</th>
                                    </tr>
                                </thead>
                                <tbody>
                                    {this.props.winners.map((winner, i) => {
                                        return (
                                            <tr key={i}>
                                                <td>{winner.year}</td>
                                                <td>{winner.name}</td>
                                            </tr>
                                        );
                                    })}
                                </tbody>
                            </table>
                        </div>
                    </Col>
                </Row>
            </div>
        );
    }
}

Award.defaultProps = {
    image: "",
    alt: "",
    title: "",
    subtitle: "",
    metatitle: "",
    winners: [
        {
            year: "",
            name: ""
        }
    ]
};

Award.propTypes = {
    image: PropTypes.string,
    alt: PropTypes.string,
    title: PropTypes.string,
    subtitle: PropTypes.string,
    metatitle: PropTypes.string,
    winners: PropTypes.array
};

export default Award;