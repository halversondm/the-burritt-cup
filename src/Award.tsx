import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import {ReactNode} from "react";

interface AwardProps {
    image: string,
    alt: string,
    title: string,
    subtitle: string,
    metatitle: string,
    winners: Winner[]
}

export interface Winner {
    year: string,
    name: string
}

export default function Award({
                                  image,
                                  alt,
                                  title,
                                  subtitle,
                                  metatitle,
                                  winners,
                              }: AwardProps): ReactNode {

    return (
        <div>
            <hr/>
            <Row className="justify-content-center">
                <Col lg={8} md={10}>
                    <div className="post-preview">
                        {image !== "" ?
                            <img className="img-responsive" src={image} alt={alt}/> : null}
                        <h2 className="post-title">{title}</h2>
                        <h4 className="post-subtitle">{subtitle}</h4>
                        <p className="post-meta">{metatitle}</p>
                        <p className="post-meta">Past Winners:</p>
                        <table className="table table-striped table-responsive">
                            <thead>
                            <tr>
                                <th>Year</th>
                                <th>Players</th>
                            </tr>
                            </thead>
                            <tbody>
                            {winners.map((winner, i) => {
                                return (
                                    <tr key={i}>
                                        <td>{winner.year}</td>
                                        <td>{winner.name}</td>
                                    </tr>
                                );
                            })}
                            </tbody>
                        </table>
                    </div>
                </Col>
            </Row>
        </div>
    );
}