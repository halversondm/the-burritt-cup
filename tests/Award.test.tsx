import React from 'react';
import {render} from '@testing-library/react'
import Award, {Winner} from '../src/Award';

test("Award component", () => {
    const winners: Winner[] = [{year: "", name: ""}]
    const unit = <Award alt="" image="" metatitle="" subtitle="" title="" winners={winners}/>;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});