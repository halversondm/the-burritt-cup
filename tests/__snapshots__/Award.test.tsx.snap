// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Award component 1`] = `"<div><div><hr><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><div class="post-preview"><h2 class="post-title"></h2><h4 class="post-subtitle"></h4><p class="post-meta"></p><p class="post-meta">Past Winners:</p><table class="table table-striped table-responsive"><thead><tr><th>Year</th><th>Players</th></tr></thead><tbody><tr><td></td><td></td></tr></tbody></table></div></div></div></div></div>"`;
